### PR TITLE
Adding configs for some PG2 compsets

### PR DESCRIPTION
**Pull request by mt5555**
new compsets:

full suport:
ne30pg2_r05_oEC60to30_ICG

preliminary support to enable further dev/testing for:  
ne256pg2_r0125_oRRS18to6v3
ne512pg2_r0125_oRRS18to6v3
ne1024pg2_r0125_oRRS18to6v3

also:
-new mapping files and domain files for ne30pg2 with improved precision (rowsums and conservation to 15 digits)
-tweaks to EAM defaults so r05 use cases dont pick up ne30 surfdata sets
-tweaks to pelayouts so ne30pg2 uses same layouts as ne30np4 compsets 

[BFB]

### CODE REVIEW

File: cime/config/e3sm/config_grids.xml [hunk 1]
```diff
@@ -2031,8 +2061,8 @@
     <domain name="ne30np4.pg2">
       <nx>21600</nx>
       <ny>1</ny>
-      <file grid="atm|lnd" mask="oEC60to30v3">$DIN_LOC_ROOT/share/domains/domain.lnd.ne30pg2_oEC60to30v3.200110.nc</file>
-      <file grid="ice|ocn" mask="oEC60to30v3">$DIN_LOC_ROOT/share/domains/domain.ocn.ne30pg2_oEC60to30v3.200110.nc</file>
+      <file grid="atm|lnd" mask="oEC60to30v3">$DIN_LOC_ROOT/share/domains/domain.lnd.ne30pg2_oEC60to30v3.200220.nc</file>
+      <file grid="ice|ocn" mask="oEC60to30v3">$DIN_LOC_ROOT/share/domains/domain.ocn.ne30pg2_oEC60to30v3.200220.nc</file>
```
Comment (oksanaguba) [inline cime/config/e3sm/config_grids.xml]: I do not see these new domain files in https://web.lcrc.anl.gov/public/e3sm/inputdata/share/domains/ or https://svn-ccsm-inputdata.cgd.ucar.edu/trunk/inputdata/share/domains/ , are they supposed to be?

File: cime/config/e3sm/config_grids.xml [hunk 2]
```diff
@@ -2113,6 +2143,34 @@
...
+      <desc>ne512np4.pg2 is Spectral Elem 6km grid w/ 2x2 FV physics grid per element:</desc>
+    </domain>
+
+    <domain name="ne1024np4.pg2">
+      <nx>25165824</nx>
+      <ny>1</ny>
+      <file grid="atm|lnd" mask="oRRS18to6v3">$DIN_LOC_ROOT/share/domains/domain.lnd.ne1024pg2_oRRS18to6v3.200212.nc</file>
+      <file grid="ice|ocn" mask="oRRS18to6v3">$DIN_LOC_ROOT/share/domains/domain.ocn.ne1024pg2_oRRS18to6v3.200212.nc</file>
+      <desc>ne1024np4.pg2 is Spectral Elem 3km grid w/ 2x2 FV physics grid per element:</desc>
+    </domain>
```
Comment (oksanaguba) [inline cime/config/e3sm/config_grids.xml]: Same about these.

File: cime/config/e3sm/config_grids.xml [hunk 3]
```diff
@@ -2587,17 +2646,17 @@
     </gridmap>
 
     <gridmap atm_grid="ne30np4.pg2" ocn_grid="oEC60to30v3_ICG">
-      <map name="ATM2OCN_FMAPNAME">cpl/gridmaps/ne30np4/map_ne30np4_to_oEC60to30v3_aave.161222.nc</map>
-      <map name="ATM2OCN_VMAPNAME">cpl/gridmaps/ne30np4/map_ne30np4_to_oEC60to30v3_conserve.161222.nc</map>
-      <map name="ATM2OCN_SMAPNAME">cpl/gridmaps/ne30np4/map_ne30np4_to_oEC60to30v3_conserve.161222.nc</map>
-      <map name="OCN2ATM_FMAPNAME">cpl/gridmaps/oEC60to30v3/map_oEC60to30v3_to_ne30np4_aave.161222.nc</map>
-      <map name="OCN2ATM_SMAPNAME">cpl/gridmaps/oEC60to30v3/map_oEC60to30v3_to_ne30np4_aave.161222.nc</map>
+      <map name="ATM2OCN_FMAPNAME">cpl/gridmaps/ne30pg2/map_ne30pg2_to_oEC60to30v3_mono.200220.nc</map>
```
Comment (oksanaguba) [inline cime/config/e3sm/config_grids.xml]: same here.

**Comment by mt5555** [comment]
most map files added to the inputdata server.

changes to ELM are to address issue #3464

File: cime/config/e3sm/config_grids.xml [hunk 4]
```diff
@@ -1023,9 +1023,9 @@
       <mask>oEC60to30v3</mask>
     </model_grid>
 
-    <model_grid alias="ne30pg2_oECv3_ICG" compset="(CAM5.+CLM.+MPASO%SPUNUP)">
+    <model_grid alias="ne30pg2_r05_oECv3_ICG" compset="(CAM5.+CLM.+MPASO%SPUNUP)">
```
Comment (brhillman) [inline cime/config/e3sm/config_grids.xml]: Any particular reason you chose to remove the configuration with atm and lnd on the same grid when adding the tri-grid? Just so that we don't have to support both?

File: cime/config/e3sm/config_grids.xml [hunk 5]
```diff
@@ -2766,6 +2825,59 @@
       <map name="LND2ATM_SMAPNAME">cpl/gridmaps/360x720cru/map_360x720cru_to_ne1024np4_mono.190508.nc</map>
     </gridmap>
 
+    <gridmap atm_grid="ne256np4.pg2" ocn_grid="oRRS18to6v3">
+      <!-- 13km atm /  18to6 ocean, so use bilin for state -->
+      <map name="ATM2OCN_FMAPNAME">cpl/gridmaps/ne256pg2/map_ne256pg2_to_oRRS18to6v3_nco.200212.nc</map>
+      <map name="ATM2OCN_SMAPNAME">cpl/gridmaps/ne256pg2/map_ne256pg2_to_oRRS18to6v3_bilin.200212.nc</map>
+      <map name="ATM2OCN_VMAPNAME">cpl/gridmaps/ne256pg2/map_ne256pg2_to_oRRS18to6v3_bilin.200212.nc</map>
+      <map name="OCN2ATM_FMAPNAME">cpl/gridmaps/ne256pg2/map_oRRS18to6v3_to_ne256pg2_nco.200212.nc</map>
+      <map name="OCN2ATM_SMAPNAME">cpl/gridmaps/ne256pg2/map_oRRS18to6v3_to_ne256pg2_nco.200212.nc</map>
```
Comment (brhillman) [inline cime/config/e3sm/config_grids.xml]: SMAP with the nco algorithm...is this because the bilin or highorder maps had bad weights?

File: cime/config/e3sm/config_grids.xml [hunk 6]
```diff
@@ -2766,6 +2825,59 @@
...
+    </gridmap>
+
+
+    <gridmap atm_grid="ne512np4.pg2" ocn_grid="oRRS18to6v3">
+      <!-- 6km atm /  18to6 ocean, so use bilin for state -->
+      <map name="ATM2OCN_FMAPNAME">cpl/gridmaps/ne512pg2/map_ne512pg2_to_oRRS18to6v3_nco.200212.nc</map>
+      <map name="ATM2OCN_SMAPNAME">cpl/gridmaps/ne512pg2/map_ne512pg2_to_oRRS18to6v3_bilin.200212.nc</map>
+      <map name="ATM2OCN_VMAPNAME">cpl/gridmaps/ne512pg2/map_ne512pg2_to_oRRS18to6v3_bilin.200212.nc</map>
+      <map name="OCN2ATM_FMAPNAME">cpl/gridmaps/ne512pg2/map_oRRS18to6v3_to_ne512pg2_nco.200212.nc</map>
+      <map name="OCN2ATM_SMAPNAME">cpl/gridmaps/ne512pg2/map_oRRS18to6v3_to_ne512pg2_nco.200212.nc</map>
```
Comment (brhillman) [inline cime/config/e3sm/config_grids.xml]: Same here, SMAPs using nco.

File: cime/config/e3sm/config_grids.xml [hunk 7]
```diff
@@ -2766,6 +2825,59 @@
...
+    </gridmap>
+
+    <gridmap atm_grid="ne512np4.pg2" lnd_grid="r0125">
+      <!-- 6km atm /  13km land.  downscale atm->land, land->atm use bilinear -->
+      <map name="ATM2LND_FMAPNAME">cpl/gridmaps/ne512pg2/map_ne512pg2_to_r0125_mono.200212.nc</map>
+      <map name="ATM2LND_SMAPNAME">cpl/gridmaps/ne512pg2/map_ne512pg2_to_r0125_mono.200212.nc</map>
```
Comment (brhillman) [inline cime/config/e3sm/config_grids.xml]: LND2ATM smap uses bilin, but ATM2LND smap uses mono?

File: cime/config/e3sm/config_grids.xml [hunk 8]
```diff
@@ -2766,6 +2825,59 @@
...
+      <map name="ATM2OCN_SMAPNAME">cpl/gridmaps/ne1024pg2/map_ne1024pg2_to_oRRS18to6v3_nco.200212.nc</map>
+      <map name="ATM2OCN_VMAPNAME">cpl/gridmaps/ne1024pg2/map_ne1024pg2_to_oRRS18to6v3_nco.200212.nc</map>
+      <map name="OCN2ATM_FMAPNAME">cpl/gridmaps/ne1024pg2/map_oRRS18to6v3_to_ne1024pg2_nco.200212.nc</map>
+      <map name="OCN2ATM_SMAPNAME">cpl/gridmaps/ne1024pg2/map_oRRS18to6v3_to_ne1024pg2_nco.200212.nc</map>
+    </gridmap>
+
+    <gridmap atm_grid="ne1024np4.pg2" lnd_grid="r0125">
+      <!-- 3km atm /  13km land.  downscale atm->land.  land->atm use bilinear-->
+      <map name="ATM2LND_FMAPNAME">cpl/gridmaps/ne1024pg2/map_ne1024pg2_to_r0125_mono.200212.nc</map>
+      <map name="ATM2LND_SMAPNAME">cpl/gridmaps/ne1024pg2/map_ne1024pg2_to_r0125_mono.200212.nc</map>
```
Comment (brhillman) [inline cime/config/e3sm/config_grids.xml]: Same as above, ATM2LND_SMAP uses mono but LND2ATM uses bilin?

File: components/cam/bld/namelist_files/namelist_defaults_cam.xml [hunk 9]
```diff
@@ -162,7 +162,12 @@
 <ncdata dyn="se" hgrid="ne240np4" nlev="26"             ic_ymd="901" >atm/cam/inic/homme/cami_0000-09-01_ne240np4_L26_c061106.nc</ncdata>
 <ncdata dyn="se" hgrid="ne512np4" nlev="72"        ic_ymd="20160801">atm/cam/inic/homme/ifs_oper_T1279_2016080100_mod_subset_to_e3sm_ne512np4_topoadj_L72.nc</ncdata>
 <ncdata dyn="se" hgrid="ne1024np4" nlev="72"       ic_ymd="20160801">atm/cam/inic/homme/ifs_oper_T1279_2016080100_mod_subset_to_e3sm_ne1024np4_topoadj_L72.nc</ncdata>
-<ncdata dyn="se" hgrid="ne1024np4" nlev="120"      ic_ymd="20160801">atm/cam/inic/homme/ifs_oper_T1279_2016080100_mod_subset_to_e3sm_ne1024np4_topoadj_L120.nc</ncdata>
+<ncdata dyn="se" hgrid="ne1024np4" nlev="120"      ic_ymd="20160801">atm/cam/inic/homme/ifs_oper_T1279_2016080100_mod_subset_to_e3sm_ne1024np4_topoadj_L120.nc</ncdata
+>
```
Comment (brhillman) [inline components/cam/bld/namelist_files/namelist_defaults_cam.xml]: This looks like a typo; `>` should be on the previous line following `ncdata`.

File: components/cam/bld/namelist_files/namelist_defaults_cam.xml [hunk 10]
```diff
@@ -162,7 +162,12 @@
...
+<ncdata dyn="se" hgrid="ne256np4"           >DUMMY-IC</ncdata>
```
Comment (brhillman) [inline components/cam/bld/namelist_files/namelist_defaults_cam.xml]: Why are these here? Shouldn't we provide ICs for these configurations?

File: components/cam/bld/namelist_files/namelist_defaults_cam.xml [hunk 11]
```diff
@@ -235,6 +240,9 @@
 <bnd_topo hgrid="ne240np4" npg="0">atm/cam/topo/USGS-gtopo30_ne240np4_16xconsistentSGH.c20130724.nc</bnd_topo>
 <bnd_topo hgrid="ne512np4"  npg="0">atm/cam/topo/USGS-gtopo30_ne512np4_16xconsistentSGH_20190212.nc</bnd_topo>
 <bnd_topo hgrid="ne1024np4" npg="0">atm/cam/topo/USGS-gtopo30_ne1024np4_16xconsistentSGH_20190528.nc</bnd_topo>
+<bnd_topo hgrid="ne256np4"  npg="2">atm/cam/topo/USGS-gtopo30_ne256np4pg2_16xdel2_20200213.nc</bnd_topo>
```
Comment (brhillman) [inline components/cam/bld/namelist_files/namelist_defaults_cam.xml]: Naming convention is different here...is this the version of the topo before the SGH has been adjusted?

File: components/cam/bld/namelist_files/namelist_defaults_cam.xml [hunk 12]
```diff
@@ -822,6 +830,10 @@
 <drydep_srf_file hgrid="ne240np4">atm/cam/chem/trop_mam/atmsrf_ne240np4_110920.nc</drydep_srf_file>
 <drydep_srf_file hgrid="ne512np4">atm/cam/chem/trop_mam/atmsrf_ne512np4_interp_20190409.nc</drydep_srf_file>
 <drydep_srf_file hgrid="ne1024np4">atm/cam/chem/trop_mam/atmsrf_ne1024np4_20190621.nc</drydep_srf_file>
+<drydep_srf_file hgrid="ne256np4"  npg="2">atm/cam/chem/trop_mam/atmsrf_ne256pg2_200212.nc</drydep_srf_file>
```
Comment (brhillman) [inline components/cam/bld/namelist_files/namelist_defaults_cam.xml]: What do you think about using the full year for the datestrings going forward? I.e., 20200212 instead of 200212?

File: components/clm/bld/CLMBuildNamelist.pm [hunk 13]
```diff
@@ -1877,7 +1877,7 @@ sub process_namelist_commandline_use_case {
     my $uc_defaults = Build::NamelistDefaults->new("$opts->{'use_case_dir'}/$opts->{'use_case'}.xml", $cfg);
 
     my %settings;
-    $settings{'res'}            = $nl_flags->{'res'};
+    $settings{'hgrid'}          = $nl_flags->{'res'};
```
Comment (brhillman) [inline components/clm/bld/CLMBuildNamelist.pm]: Showing my ignorance regarding the CLMBuildNamelist script, but why did this need to change?

File: components/cam/bld/namelist_files/namelist_defaults_cam.xml (hunk 11 above)
Comment (ambrad) [reply components/cam/bld/namelist_files/namelist_defaults_cam.xml]: I created the ne256np4pg2 topo file from scratch, following workflow 2 on the docs page. For now, ne512 and ne1024 np4pg2 were created using the converter, following workflow 1 on the docs page. The names have different conventions to reflect the different workflows I used.

File: cime/config/e3sm/config_grids.xml (hunk 4 above)
Comment (mt5555) [reply cime/config/e3sm/config_grids.xml]: Yes, I thought for V2 we will only be supporting land on r05 and r0125 grids.

File: cime/config/e3sm/config_grids.xml (hunk 5 above)
Comment (mt5555) [reply cime/config/e3sm/config_grids.xml]: currently, all ocn->atm maps have to use the same algorithm due to a technical issue with different range (mapped target points) of different algorithms.  we might be able to relax this in the future, but not now.

File: cime/config/e3sm/config_grids.xml (hunk 6 above)
Comment (mt5555) [reply cime/config/e3sm/config_grids.xml]: yes, same reason as above.

File: cime/config/e3sm/config_grids.xml (hunk 7 above)
Comment (mt5555) [reply cime/config/e3sm/config_grids.xml]: when the source grid is higher resolution than the target grid, then "bilin" is a bad choice, and the mono/aave algorithms are actually better.

File: cime/config/e3sm/config_grids.xml (hunk 8 above)
Comment (mt5555) [reply cime/config/e3sm/config_grids.xml]: as above, atm resolution >> lnd resolution

File: components/clm/bld/CLMBuildNamelist.pm (hunk 13 above)
Comment (mt5555) [reply components/clm/bld/CLMBuildNamelist.pm]: it's a bug - see  issue #3464

File: components/cam/bld/namelist_files/namelist_defaults_cam.xml (hunk 12 above)
Comment (mt5555) [reply components/cam/bld/namelist_files/namelist_defaults_cam.xml]: no preference. it seems like both formats are common.

File: components/cam/bld/namelist_files/namelist_defaults_cam.xml (hunk 10 above)
Comment (mt5555) [reply components/cam/bld/namelist_files/namelist_defaults_cam.xml]: for the SCREAM resolutions, I thought we should wait until the SCREAM physics is closer to being finalized.

File: components/cam/bld/namelist_files/namelist_defaults_cam.xml (hunk 9 above)
Comment (mt5555) [reply components/cam/bld/namelist_files/namelist_defaults_cam.xml]: will fix.

**Comment by golaz** [comment]
@whannah1 : do you want to take a look at these changes?
@oksanaguba : is this otherwise ready to be merged?

**Comment by oksanaguba** [comment]
I will look at this today, assuming everyone else is done with this PR. thanks!

**Comment by mt5555** [comment]
yes, this is ready to go and should be merged ASAP

File: cime/config/e3sm/config_grids.xml [hunk 14]
```diff
@@ -2766,6 +2825,59 @@
       <map name="LND2ATM_SMAPNAME">cpl/gridmaps/360x720cru/map_360x720cru_to_ne1024np4_mono.190508.nc</map>
     </gridmap>
 
+    <gridmap atm_grid="ne256np4.pg2" ocn_grid="oRRS18to6v3">
+      <!-- 13km atm /  18to6 ocean, so use bilin for state -->
+      <map name="ATM2OCN_FMAPNAME">cpl/gridmaps/ne256pg2/map_ne256pg2_to_oRRS18to6v3_nco.200212.nc</map>
+      <map name="ATM2OCN_SMAPNAME">cpl/gridmaps/ne256pg2/map_ne256pg2_to_oRRS18to6v3_bilin.200212.nc</map>
```
Comment (oksanaguba) [inline cime/config/e3sm/config_grids.xml]: I don't see cpl/gridmaps/ne256pg2/map_ne256pg2_to_oRRS18to6v3_bilin.200212.nc in the repo:
https://web.lcrc.anl.gov/public/e3sm/inputdata/cpl/gridmaps/ne256pg2/

Comment (mt5555) [reply cime/config/e3sm/config_grids.xml]: the ne256pg2, ne512pg2 andne1024pg2 cases are not finalized yet.  The are also missing initial condition files.   so those are not expected to work out of the box at this time.

Comment (oksanaguba) [reply cime/config/e3sm/config_grids.xml]: Can this be stated clearly in the description? It says 'new compsets'... So, only ne30 compset would work? What is the line to config this compset? I want to try.

Comment (mt5555) [reply cime/config/e3sm/config_grids.xml]: updated.

**Comment by oksanaguba** [comment]
Is it really nonBFB for nightlies? We don't have ne30pg2 F or coupled cases in nightlies, or do we?

File: components/cam/bld/namelist_files/namelist_defaults_cam.xml (hunk 9 above)
Comment (brhillman) [reply components/cam/bld/namelist_files/namelist_defaults_cam.xml]: This looks like it still needs to be fixed?

Comment (oksanaguba) [reply components/cam/bld/namelist_files/namelist_defaults_cam.xml]: ... and pushed to next...

Comment (oksanaguba) [reply components/cam/bld/namelist_files/namelist_defaults_cam.xml]: will do this now

Comment (oksanaguba) [reply components/cam/bld/namelist_files/namelist_defaults_cam.xml]: I think I am viewing outdated file somehow -- this is fixed in the branch.

File: components/cam/bld/namelist_files/namelist_defaults_cam.xml (hunk 10 above)
Comment (brhillman) [reply components/cam/bld/namelist_files/namelist_defaults_cam.xml]: That's fine. I do think we include them in the SCREAM repo though, but I guess we don't have to include them here. These resolutions are run at your own risk anyways I guess.
